File: lesson_20/src/main/java/lesson20/entities/Customer.java
package lesson20.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "customer")
//пользователи (id, имя, возраст)
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    Long id;
    @Column(name = "fio")
    String fio;
    @Column(name = "age")
    Integer age;

    public Long getId() {
        return id;
    }

    @Override
    public String toString() {
        return "Customer{" +
                "id=" + id +
                ", fio='" + fio + '\'' +
                ", age=" + age +
                '}';
    }

    public String getFio() {
        return fio;
    }

    public void setFio(String fio) {
        this.fio = fio;
    }

    public Integer getAge() {
        return age;
    }

    public void setAge(Integer age) {
        this.age = age;
    }

    public Customer(String fio, Integer age) {
        this.fio = fio;
        this.age = age;
    }

    public Customer() {
    }
}

File: lesson_22/src/main/kotlin/com/lesson22/entities/Customer.kt
package com.lesson22.entities

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "customer")
data class Customer (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    var id: Long? = null,
    @Column(name = "fio")
    var fio: String? = null,
    @Column(name = "age")
    var age: Int? = null,
    //@Type(type = "jsonb")
    @Column(name = "test", columnDefinition = "jsonb")
    var test: MyType? = null
)

data class MyType (
    var long: Long? = null,
    var string: String? = null,
    var string2: String? = null
)
